hazelcast-jet-core/src/main/java/com/hazelcast/jet/Traversers.java

```diff
@@ -75,26 +72,27 @@ private Traversers() {
      * and return it.
      */
     @Nonnull
-    public static <T> Traverser<T> enumerate(@Nonnull Enumeration<T> enumeration) {
+    public static <T> Traverser<T> traverseEnumeration(@Nonnull Enumeration<T> enumeration) {
         return () -> enumeration.hasMoreElements() ? enumeration.nextElement() : null;
     }
 
     /**
-     * Returns a traverser over the given stream. The stream is traversed through
-     * its spliterator, which is obtained immediately.
+     * Returns a traverser over the given stream of non-null elements. It will
+     * traverse the stream through its spliterator, which it obtains
+     * immediately. When it exhausts the stream, it closes it.
```

Comment:
I'd make the non-null requirement more obvious, similar to wording in `traverseSpliterator`: 
"Returns a traverser over the given stream. (...) The stream must not emit {@code null} items."